### PR TITLE
added magnet functionality

### PR DESCRIPTION
**Pull request by Will-Cross1**
made it so every second in use it subtracts a point

drags the treasure

decreased the value of the treasure because its now slightly easier

means treasure can no longer be stuck

### CODE REVIEW

File: maze-game/Player.java [hunk 1]
```diff
@@ -467,6 +531,8 @@ private void endGame()
         world.showText("Score:" + score, 575, 400);
         world.showText("Time:" + time, 575, 450);
         
+        world.showText(null, 1005, 920);
```
Comment (rowan04) [inline maze-game/Player.java:534]: what does showText(null) do?

Comment (Will-Cross1) [reply maze-game/Player.java:534]: removes the text from the screen

File: maze-game/Player.java [hunk 2]
```diff
@@ -412,6 +451,30 @@ private void resetSpeed()
...
         activateSpeedPotion = false;
     }
+    
+    private void toggleMagnet()
+    {
+        if(magState == false)
+        {
+            if(MyWorld.score != 0)
+            {
+                MyWorld.score -= 1;
```
Comment (rowan04) [inline maze-game/Player.java:461]: why is the score reduced here

Comment (Will-Cross1) [reply maze-game/Player.java:461]: so that they can't keep spamming m and using the magnet for free

File: maze-game/Player.java [hunk 3]
```diff
@@ -290,6 +311,24 @@ private void collectZapper()
         
         Greenfoot.playSound("ghost_busters.mp3");
     }
+    
+    /**
+     * if player is touching Zapper, collect it, then delete the Zapper object
```
Comment (rowan04) [inline maze-game/Player.java:316]:
```suggestion
     * if player is touching magnet, collect it, then delete the magnet object
```

File: maze-game/Player.java [hunk 4]
```diff
@@ -204,7 +217,15 @@ public void act()
         }
         
         /**
-         * if touching the teleporter, call the collectTeleporter function
+         * if touching the teleporter, call the collectMagnet function
```
Comment (rowan04) [inline maze-game/Player.java:220]:
```suggestion
         * if touching the magnet, call the collectMagnet function
```

File: maze-game/Player.java [hunk 5]
```diff
@@ -204,7 +217,15 @@ public void act()
...
+         */
+        if (isTouching(Magnet.class))
+        {
+            collectMagnet();
+        }
+        
+        /**
+         * if touching the magnet, call the collectTeleporter function
```
Comment (rowan04) [inline maze-game/Player.java:228]:
```suggestion
         * if touching the teleporter, call the collectTeleporter function
```

File: maze-game/Magnet.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+ * Write a description of class Magnet here.
+ * 
+ * @author (your name) 
+ * @version (a version number or a date)
+ */
+public class Magnet extends Actor
+{
+    public Magnet()
+    {
+        GreenfootImage beam = getImage();
```
Comment (rowan04) [inline maze-game/Magnet.java:13]:
```suggestion
        GreenfootImage magnet = getImage();
```

File: maze-game/Magnet.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        beam.scale(50, 50);
```
Comment (rowan04) [inline maze-game/Magnet.java:14]:
```suggestion
        magnet.scale(50, 50);
```

File: maze-game/Player.java (hunk 2 above)
Comment (rowan04) [reply maze-game/Player.java:461]: oh

File: maze-game/Player.java (hunk 1 above)
Comment (rowan04) [reply maze-game/Player.java:534]: ok

**Comment by Will-Cross1** [comment]
too late

**Comment by rowan04** [comment]
i mean its hard to see if it works or not just from text. might as well just merge it, we'll see if any problems occur soon enough

**Review by rowan04 — APPROVED** [review]

**Pull request merged**